clang/lib/DPCT/DPCT.cpp

```diff
@@ -736,8 +739,33 @@ int runDPCT(int argc, const char **argv) {
   CudaPath = getCudaInstallPath(OriginalArgc, argv);
   DpctDiags() << "Cuda Include Path found: " << CudaPath << "\n";
 
-  RefactoringTool Tool(OptParser->getCompilations(),
-                       OptParser->getSourcePathList());
+  std::vector<std::string> SourcePathList;
+  if (QueryAPIMapping.getNumOccurrences()) {
+    APIMapping::initEntryMap();
+    auto SourceCode = APIMapping::getAPISourceCode(QueryAPIMapping);
+    if (SourceCode.empty()) {
+      llvm::outs() << "The API Mapping is not available\n";
+      dpctExit(MigrationSucceeded);
+    }
+    llvm::outs() << "CUDA API:" << getApiMappingStr(SourceCode);
+    int FD;
+    SmallString<64> TempFile;
+    if (auto EC =
+            llvm::sys::fs::createTemporaryFile("tmp", "cu", FD, TempFile)) {
+      llvm::errs() << "Error: dpct internal error. Cannot create temp file for "
+                      "api mapping.\n";
+    }
```

Comment:
if failed to create temp file, then exit the dpct program.

Reply:
done